torch/_inductor/codegen/common.py

```diff
@@ -1023,7 +1027,15 @@ def indirect_indexing(var, size, check=True):
                     if var.bounds.upper >= 0:
                         lt = ops.lt(var, "0")
                         stm = ops.where(lt, stm, var)
-                    new_var = self.cse.generate(self.compute, stm, bounds=new_bounds)
+
+                    new_var = self.cse.generate(
+                        self.compute,
+                        stm,
+                        bounds=new_bounds,
+                        # Old and new bounds may be non-intersecting
+                        # Just use the new bounds if it is known
+                        ignore_old_bound=new_bounds != ValueRanges.unknown(),
```

Comment:
Note there is a separate bug even if we do fallback. The new bounds and old bounds can be non intersecting.

Reply:
Maybe this can be handled separately to @guilhermeleobas's work @lezcano

Reply:
yep, he'll submit a fix with this (he had already separately found this bug, he's just putting together a comprehensive fix for all these issues)